crates/solvers/src/boundary/naive.rs

```diff
@@ -0,0 +1,245 @@
+                                ..Default::default()
+                            }),
+                            order::Class::Liquidity => OrderClass::Liquidity,
+                        },
+                        solver_fee: order.fee().amount,
+                        ..Default::default()
+                    },
+                    data: OrderData {
+                        sell_token: order.sell.token.0,
+                        buy_token: order.buy.token.0,
+                        sell_amount: order.sell.amount,
+                        buy_amount: order.buy.amount,
+                        fee_amount: order.fee().amount,
+                        kind: match order.side {
+                            order::Side::Buy => OrderKind::Buy,
+                            order::Side::Sell => OrderKind::Sell,
+                        },
+                        partially_fillable: order.partially_fillable,
+                        ..Default::default()
+                    },
+                    ..Default::default()
+                },
+            }),
+            exchange: Exchange::GnosisProtocol,
+            reward: 0.,
+        })
+        .collect_vec();
+
+    let slippage = Slippage::new(pool.tokens());
+    let pool_handler = Arc::new(PoolHandler::default());
+    let boundary_pool = ConstantProductOrder::for_pool(
+        to_boundary_pool(liquidity.address, pool)?,
+        pool_handler.clone(),
+    );
+
+    let boundary_solution =
+        multi_order_solver::solve(&slippage.context(), boundary_orders, &boundary_pool)?;
+
+    let swap = pool_handler.swap.lock().unwrap().take();
+    Some(solution::Solution {
+        prices: solution::ClearingPrices::new(
+            boundary_solution
+                .clearing_prices()
+                .iter()
+                .map(|(token, price)| (eth::TokenAddress(*token), *price)),
+        ),
+        trades: boundary_solution
+            .traded_orders()
+            .map(|order| {
+                solution::Trade::Fulfillment(solution::Fulfillment::fill(
+                    orders
+                        .iter()
+                        .copied()
+                        .find(|o| o.uid.0 == order.metadata.uid.0)
+                        .unwrap()
+                        .clone(),
+                ))
+            })
+            .collect(),
+        interactions: swap
+            .into_iter()
+            .map(|(input, output)| {
+                solution::Interaction::Liquidity(solution::LiquidityInteraction {
+                    liquidity: liquidity.clone(),
+                    input,
+                    output,
+                    internalize: false,
+                })
+            })
+            .collect(),
+    })
+}
+
+// Beyond this point is... well... nameless and boundless chaos. The
+// unfathomable horrors that follow are not for the faint of heart!
+//
+// Joking asside, the existing naive solver implementation is tightly coupled
```

Comment:
```suggestion
// Joking aside, the existing naive solver implementation is tightly coupled
```

Whatever happened about that spell checker?